sklearn/metrics/tests/test_ranking.py

```diff
@@ -671,6 +671,15 @@ def test_binary_clf_curve():
         precision_recall_curve(y_true, y_pred)
 
 
+def test_binary_clf_curve_implicit_pos_label():
+    y_true = ["a", "b"]
+    y_pred = [0., 1.]
+    msg = ("make y_true take integer value in {0, 1} or {-1, 1}"
+           " or pass pos_label explicitly.")
+    with pytest.raises(ValueError, match=msg):
+        precision_recall_curve(y_true, y_pred)
```

Comment:
@qinhanmin2014  I have changed the to trigger the exception in `_binary_clf_curve`  from `precision_recall_curve` rather then `plot_roc_curve` so that we can merge both #15316  while also improving the error message.

Reply:
This PR is still useful to:

- improve the error message when passing string labels;
- have more tests.